internal/controller/object/object.go

```diff
@@ -358,6 +358,16 @@ func setObserved(obj *v1alpha1.Object, observed *unstructured.Unstructured) erro
 	if obj.Status.AtProvider.Manifest.Raw, err = observed.MarshalJSON(); err != nil {
 		return errors.Wrap(err, errFailedToMarshalExisting)
 	}
+
+	if obj.Spec.Readiness.Policy == v1alpha1.ReadinessPolicyDeriveFromObject {
+		conditioned := xpv1.ConditionedStatus{}
+		err := fieldpath.Pave(observed.Object).GetValueInto("status", &conditioned)
```

Comment:
Could logging the error be helpful for debugging purposes? I see a couple of cases that might cause an error here, and seeing it in the logs or events would be beneficial rather than just seeing the resource as unavailable no matter what.